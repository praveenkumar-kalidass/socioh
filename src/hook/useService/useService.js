import * as Keychain from 'react-native-keychain';
import AsyncStorage from '@react-native-community/async-storage';
import RNContacts from 'react-native-contacts';

import { CONSTANT, ERROR, TRANSLATION } from '../../constant';
import useAjax from '../useAjax';
import useUser from '../useUser';

const useService = () => {
  const { ajax } = useAjax();
  const { user, saveUserDetails, clearUserDetails } = useUser();

  const signUp = async ({ name, email, password }) => {
    await Keychain.resetGenericPassword();
    await Keychain.setGenericPassword(email, password);
    await AsyncStorage.setItem(
      CONSTANT.ASYNC_STORAGE_KEY.SOCIOH_USER,
      JSON.stringify({
        name,
        email,
      }),
    );
    saveUserDetails({
      name,
      email,
    });
    return ajax();
  };

  const signIn = async ({ email, password }) => {
    const credentials = await Keychain.getGenericPassword();
    if (credentials.username === email && credentials.password === password) {
      const userDetails = await AsyncStorage.getItem(
        CONSTANT.ASYNC_STORAGE_KEY.SOCIOH_USER,
      );
      saveUserDetails(JSON.parse(userDetails));
      return ajax();
    }
    throw Error(ERROR.INVALID_CREDENTIALS);
  };

  const logout = async () => {
    clearUserDetails();
    return ajax();
  };

  const getUserDetails = async () => {
    if (user) {
      await ajax();
      return {
        ...user,
        interests: Object.values(TRANSLATION.INTERESTS),
      };
    }
    throw Error(ERROR.NO_USER_FOUND);
  };

  const getContacts = async (search) => {
    const permission = await RNContacts.requestPermission();
    if (permission === CONSTANT.PERMISSION.AUTHORIZED) {
      let contacts;
      if (search) {
        contacts = await RNContacts.getContactsMatchingString(search);
      } else {
        contacts = await RNContacts.getAll();
      }
      await ajax();
      return contacts;
    }
    throw Error(ERROR.UNAUTHORIZED_TO_READ_CONTACTS);
  };

  const getMessages = async () => {
    await ajax();
    return CONSTANT.MESSAGES.GREETINGS;
  };

  const getMessageList = async () => {
    await ajax();
    return CONSTANT.MESSAGES.LIST;
  };

  return {
    signUp,
    signIn,
    logout,
    getUserDetails,
    getContacts,
    getMessages,
    getMessageList,
  };
};

export default useService;
