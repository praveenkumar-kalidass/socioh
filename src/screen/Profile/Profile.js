import React from 'react';
import { Card, CardItem, Thumbnail } from 'native-base';

import User from '../../asset/image/user.png';
import { CONSTANT, TRANSLATION } from '../../constant';
import wrapPattern from '../../helper/wrapPattern';
import {
  ProfileContainer,
  AvatarContainer,
  AvatarLabel,
  AvatarDescription,
  InterestsContainer,
  InterestsTitleContainer,
  InterestsTitle,
  BadgeContainer,
  InterestChip,
  Interest,
} from './Profile.style';

const Profile = () => {
  return (
    <ProfileContainer>
      <Card>
        <CardItem>
          <AvatarContainer>
            <Thumbnail large source={User} />
            <AvatarLabel bold>Sample Text</AvatarLabel>
            <AvatarDescription>
              {TRANSLATION.I_AM_SOCIOH_USER}
            </AvatarDescription>
          </AvatarContainer>
        </CardItem>
        <CardItem>
          <InterestsContainer>
            <InterestsTitleContainer>
              <InterestsTitle>{TRANSLATION.MY_INTERESTS}</InterestsTitle>
            </InterestsTitleContainer>
            <BadgeContainer>
              <InterestChip>
                <Interest>music</Interest>
              </InterestChip>
            </BadgeContainer>
          </InterestsContainer>
        </CardItem>
      </Card>
    </ProfileContainer>
  );
};

export default wrapPattern({
  Component: Profile,
  pattern: CONSTANT.SCREEN_BACKGROUND.BACKGROUND_2,
});
