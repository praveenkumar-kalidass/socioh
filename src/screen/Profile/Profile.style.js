import styled from 'styled-components/native';
import { Badge, Text } from 'native-base';

import { COLOR, FONT, SPACING } from 'constant';

export const ProfileContainer = styled.View`
  flex: 1;
  justify-content: center;
  padding-horizontal: ${SPACING.SPACE_20};
`;

export const AvatarContainer = styled.View`
  align-items: center;
  flex: 1;
  padding-vertical: ${SPACING.SPACE_20};
`;

export const AvatarLabel = styled(Text)`
  color: ${COLOR.BLACK};
  font-size: ${FONT.SIZE_24};
  margin-top: ${SPACING.SPACE_10};
`;

export const AvatarDescription = styled(Text)`
  color: ${COLOR.BLACK};
  font-size: ${FONT.SIZE_16};
  margin-top: ${SPACING.SPACE_10};
`;

export const InterestsContainer = styled.View`
  border-top-width: 1px;
  border-top-color: ${COLOR.PRIMARY};
  flex: 1;
  padding-top: ${SPACING.SPACE_20};
`;

export const InterestsTitleContainer = styled.View`
  align-items: center;
  margin-bottom: ${SPACING.SPACE_20};
`;

export const InterestsTitle = styled(Text)`
  color: ${COLOR.BLACK};
  font-size: ${FONT.SIZE_16};
  font-weight: bold;
`;

export const BadgeContainer = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
`;

export const InterestChip = styled(Badge)`
  background-color: ${COLOR.WHITE};
  border: 1px solid ${COLOR.SECONDARY};
  margin-bottom: ${SPACING.SPACE_10};
  margin-left: ${SPACING.SPACE_5};
  margin-right: ${SPACING.SPACE_5};
`;

export const Interest = styled(Text)`
  color: ${COLOR.BLACK};
  font-style: italic;
`;
