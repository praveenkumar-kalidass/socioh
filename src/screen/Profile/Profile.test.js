import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import useService from '../../hook/useService';
import Profile from './index';

const serviceMock = {
  getUserDetails: jest.fn(),
};

jest.mock('../../hook/useService');

describe('Profile', () => {
  const mockNavigation = { navigate: jest.fn() };

  beforeEach(() => {
    serviceMock.getUserDetails.mockResolvedValueOnce({
      name: 'Praveen',
      interests: ['music', 'food', 'sports'],
    });
    useService.mockImplementation(() => serviceMock);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<Profile />);

    expect(container).toMatchSnapshot();
  });

  it('should load user details and interests', async () => {
    const { getByTestId } = render(<Profile navigation={mockNavigation} />);

    await waitFor(() => getByTestId('profile_user_name'));

    expect(getByTestId('profile_user_name').props.children).toStrictEqual(
      'Praveen',
    );
    expect(getByTestId('profile_interest_food').props.children).toStrictEqual(
      'food',
    );
  });
});
