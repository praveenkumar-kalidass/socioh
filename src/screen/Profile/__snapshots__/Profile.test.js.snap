// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should match snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={1334}
      bbWidth={750}
      focusable={false}
      height={1334}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 1334,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={666.25}
          cy={333.5}
          fill={4294892136}
          propList={
            Array [
              "fill",
            ]
          }
          r={125}
        />
        <RNSVGCircle
          cx={83.75}
          cy={762.2857142857143}
          fill={4279943103}
          propList={
            Array [
              "fill",
            ]
          }
          r={93.75}
        />
        <RNSVGCircle
          cx={650}
          cy={1190.6000000000001}
          fill={4279943103}
          propList={
            Array [
              "fill",
            ]
          }
          r={75}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>,
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingVertical": 20,
              },
            ]
          }
        >
          <Image
            large={true}
            source={
              Object {
                "testUri": "../../../src/asset/image/user.png",
              }
            }
            style={
              Object {
                "borderRadius": 40,
                "height": 80,
                "width": 80,
              }
            }
          />
          <Text
            bold={true}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "#000000",
                  "fontSize": 24,
                  "marginTop": 10,
                },
              ]
            }
          >
            Sample Text
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "#000000",
                  "fontSize": 16,
                  "marginTop": 10,
                },
              ]
            }
          >
            Hey there! I am a socioh user.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderTopColor": "#1abfbf",
                "borderTopWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "paddingTop": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              My interests
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "backgroundColor": "#ED1727",
                    "borderRadius": 13.5,
                    "height": 27,
                    "justifyContent": "center",
                    "paddingBottom": 3,
                    "paddingLeft": 6,
                    "paddingRight": 6,
                    "paddingTop": 3,
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "borderColor": "#feda68",
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "marginBottom": 10,
                    "marginLeft": 5,
                    "marginRight": 5,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "lineHeight": 19,
                      "paddingLeft": 3,
                      "paddingRight": 3,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#000000",
                      "fontStyle": "italic",
                    },
                  ]
                }
              >
                music
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
